src/services/expense.service.js

```diff
@@ -0,0 +1,74 @@
+const { Op } = require('sequelize');
+const { Expense } = require('../models/Expense.model.js');
+
+const getAllExpense = async ({ userId, categories, from, to }) => {
+  const params = {};
+
+  if (userId) {
+    params.userId = userId;
+  }
+
+  if (categories) {
+    if (Array.isArray(categories)) {
+      params.category = { [Op.in]: categories };
+    } else {
+      params.category = categories;
+    }
+  }
+
+  if (from || to) {
+    params.spentAt = {};
+
+    if (from) {
+      params.spentAt[Op.gte] = from;
+    }
+
+    if (to) {
+      params.spentAt[Op.lte] = to;
+    }
+  }
+
+  const expList = Expense.findAll({ where: params });
```

Comment:
Consider using `await` with `Expense.findAll` to ensure that the promise is resolved before returning the result. This will prevent potential issues with asynchronous execution.